### PR TITLE
MW 361/Created-component-icon

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/utils/markdown/Markdown.module.scss [hunk 1]
```diff
@@ -9,8 +9,8 @@
 }
 
 .markdownImg {
-  max-width: 100%;
-  max-height: 100%;
+  max-width: $minWidthBlock;
+  max-height: $minWidthBlock;
```
Comment (tritonJS826) [inline mw-webapp/src/utils/markdown/Markdown.module.scss:13]: Why height is  min width?

File: mw-webapp/src/component/icon/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+/**
+ * Icon
+ */
+export const Icon = (props: IconProps) => {
+  const {iconName} = props;
+
+  const Icons: Record<IconProps["iconName"], JSX.Element> = {
+    EyeOpenedIcon: <EyeOpenedIcon {...props} />,
```
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/component/icon/Icon.tsx]: Refactored

File: mw-webapp/src/component/header/Header.module.scss [hunk 3]
```diff
@@ -31,7 +31,7 @@
   cursor: pointer;
 }
 
-@media (max-width: 600px) {
+@media (max-width: $widthMedium) {
```
Comment (tritonJS826) [inline mw-webapp/src/component/header/Header.module.scss:34]: brealpoints for the diffetent devices

File: mw-webapp/src/component/icon/Icon.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  /**
+   * Eye slash icon
+   */
+  EyeSlashedIcon: (params: IconProps) => <EyeSlashedIcon {...params} />,
+};
+
+/**
+ * Icon className
+ */
+export enum IconClassName {
```
Comment (tritonJS826) [inline mw-webapp/src/component/icon/Icon.tsx:23]: Icon type

File: mw-webapp/src/component/icon/Icon.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+   * @default IconClassName.MEDIUM
+   */
+  className: IconClassName;
+
+}
+
+/**
+ * Icon
+ */
+export const Icon = (props: IconProps) => IconDictionary[props.iconName](props);
```
Comment (tritonJS826) [inline mw-webapp/src/component/icon/Icon.tsx:49]:
```suggestion
export const Icon = (props: IconProps) => {
const className = clsx(
    scss.icon,
    props.size === IconSize.MEDIUM ? scss.medium : scss.small, // or create function get sizeClassName
 );

return IconDictionary[props.iconName]({...props, className});
```

File: mw-webapp/src/component/scrollableBlock/ScrollableBlock.module.scss [hunk 6]
```diff
@@ -1,6 +1,10 @@
-.scrollableBlock {
+.wrapper {
   position: relative;
+}
+
+.scrollableBlock {
   overflow: auto;
   width: 100%;
+  max-width: 1800px;
```
Comment (tritonJS826) [inline mw-webapp/src/component/scrollableBlock/ScrollableBlock.module.scss:8]: why 1800? why not global variable
shold it be breakpoint?
do we need it?

File: mw-webapp/src/component/scrollableBlock/ScrollableBlock.tsx [hunk 7]
```diff
@@ -28,11 +28,13 @@ interface ScrollableBlockProps {
  */
 export const ScrollableBlock = (props: PropsWithChildren<ScrollableBlockProps>) => {
   return (
-    <div
-      style={{width: `${props.width}`, height: `${props.height}`}}
-      className={clsx(styles.scrollableBlock, props.className)}
-    >
-      {props.children}
+    <div className={styles.wrapper}>
```
Comment (tritonJS826) [inline mw-webapp/src/component/scrollableBlock/ScrollableBlock.tsx:31]: comment why do we need so many wrappers

File: mw-webapp/src/component/tooltip/Tooltip.module.scss [hunk 8]
```diff
@@ -24,8 +22,8 @@ th .wrapper {
   display: flex;
   align-items: center;
   &:hover + .tooltip {
-    display: block;
     display: inline-block;
+    max-width: $maxWidthBlock;
   }
```
Comment (tritonJS826) [inline mw-webapp/src/component/tooltip/Tooltip.module.scss:27]: move to the appropriate component

File: mw-webapp/src/logic/waysTable/waysColumns.tsx [hunk 9]
```diff
@@ -109,7 +106,18 @@ export const waysColumns = [
...
   columnHelper.accessor("name", {
-    header: "Way",
+
+    /**
+     * Header
+     */
+    header: () => (
+      <Tooltip
+        position={PositionTooltip.TOP}
+        content="Way's name and it's goal"
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/waysColumns.tsx:116]:
```suggestion
        content="Way's name and goal"
```

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/component/tooltip/Tooltip.module.scss [hunk 10]
```diff
@@ -35,14 +33,19 @@ th .wrapper {
 }
 
 .right {
-  margin-top: -$marginSmall;
-  transform: translateX(100%);
+  margin-left: $tooltipMarginLeft;
+  transform: translateY(-100%);
 }
```
Comment (tritonJS826) [inline mw-webapp/src/component/tooltip/Tooltip.module.scss:38]: why do you need both translater and margin

File: mw-webapp/src/styles/_variables.scss [hunk 11]
```diff
@@ -28,8 +28,14 @@ $gapBig: 30px;
...
 $iconSizeSmall: 15px;
+$iconSizeMedium: 25px;
 $minWidthBlock: 360px;
+$minHeightBlock: 360px;
 $maxWidthBlock: 600px;
 $minWidthDate: 80px;
 $heightH2: 27px;
 $heightH3: 22px;
+
+// MediaQuery max-widths
```
Comment (tritonJS826) [inline mw-webapp/src/styles/_variables.scss:39]: !

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/component/tooltip/Tooltip.module.scss (hunk 10 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/tooltip/Tooltip.module.scss]: Deleted margin

**Pull request merged**